lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Random.hs

```diff
@@ -202,14 +203,16 @@ instance RndStateLike (RndState n) where
 -- decoded as a Byron HD random address, and where the wallet key can be used
 -- to decrypt the address derivation path.
 instance IsOurs (RndState n) Address where
+    type DerivationPath (RndState n) Address = DerivationPath
     isOurs addr st =
-        (isJust path, maybe id (addDiscoveredAddress addr Used) path st)
+        (path, maybe id (addDiscoveredAddress addr Used) path st)
```

Comment:
For random addresses, this is trivially achieved because decrypting the path is exactly how we identify whether a random address is ours or not.